### PR TITLE
Review

### PR DESCRIPTION
**Pull request by dsznajder**

### CODE REVIEW

File: lib/figure/base.rb [hunk 1]
```diff
@@ -24,6 +24,10 @@ def possible_moves
 
     private
 
+    def vectors
```
Comment (Renerte) [inline lib/figure/base.rb]: Do Not Repeat Yourself was applied here.

File: lib/figure/bishop.rb [hunk 2]
```diff
@@ -2,13 +2,12 @@
 
 module Figure
   class Bishop < Base
```
Comment (Renerte) [inline lib/figure/bishop.rb]: This class is an example of Single Responsibility Principle.

File: lib/figure/pawn.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+require 'figure/base'
+
+module Figure
+  class Pawn < Base
```
Comment (Renerte) [inline lib/figure/pawn.rb]: Do Not Repeat Yourself ;)

File: lib/figure/base.rb [hunk 4]
```diff
@@ -1,4 +1,35 @@
...
+    attr_reader :x, :y
+
+    include Support::TransformPosition
+    SCOPE = (1..8)
+    NORTH_SOUTH = [[0, 1], [0, -1]].freeze
+    WEST_EAST = [[-1, 0], [1, 0]].freeze
+    N_EAST_S_WEST = [[-1, -1], [1, 1]].freeze
+    N_WEST_S_EAST = [[-1, 1], [1, -1]].freeze
+
+    def initialize(position, properties = {})
```
Comment (Renerte) [inline lib/figure/base.rb]: Law of Demeter.

File: lib/figure/knight.rb [hunk 5]
```diff
@@ -2,13 +2,15 @@
 
 module Figure
   class Knight < Base
```
Comment (Renerte) [inline lib/figure/knight.rb]: Interface Segregation.

File: bin/run [hunk 6]
```diff
@@ -35,7 +35,7 @@ class Exercise
   def calculate_possible_positions(line)
     ChessPieceMove.new(line, figure_move_class).possible_positions
   end
-
 end
 
-Exercise.new(Figure::Queen).call
\ No newline at end of file
+p ARGV[1]
```
Comment (cs3b) [inline bin/run]: keep in mind to clean up this code :-)

File: bin/run [hunk 7]
```diff
@@ -35,7 +35,7 @@ class Exercise
...
+Exercise.new(Object.const_get('Figure').const_get(ARGV[1])).call
```
Comment (cs3b) [inline bin/run]: Isn't that line overcomplicated?  What about:

* creating support module `support/fetch_figure_class`
* on that module implement method figure_move_class
* remove that param from initializer & cleanup code in Exercise class

File: lib/chess_piece_move.rb [hunk 8]
```diff
@@ -12,6 +11,6 @@ def possible_positions
   private
 
   def possible_coordinates(coordinate)
-    figure_move_class.new(coordinate).possible_coordinates
+    figure_move_class.new(coordinate, ARGV[2]).possible_moves
```
Comment (cs3b) [inline lib/chess_piece_move.rb]: what is ARGV[2] ?

File: lib/figure/base.rb [hunk 9]
```diff
@@ -1,4 +1,35 @@
...
+
+    include Support::TransformPosition
+    include Support::Board
+    NORTH_SOUTH = [[0, 1], [0, -1]].freeze
+    WEST_EAST = [[-1, 0], [1, 0]].freeze
+    N_EAST_S_WEST = [[-1, -1], [1, 1]].freeze
+    N_WEST_S_EAST = [[-1, 1], [1, -1]].freeze
+
+    def initialize(position, color = 'white')
+      @x = position.first
```
Comment (cs3b) [inline lib/figure/base.rb]: alternative

```ruby
@x, @y = position
```

File: spec/lib/figure/rook_spec.rb [hunk 10]
```diff
@@ -1,8 +1,17 @@
...
+
+  describe '#moves' do
+    let(:rook_moves) do
+      [[3, 2], [1, 2], [2, 3], [2, 1], [4, 2], [0, 2], [2, 4], [2, 0], [5, 2],
+       [-1, 2], [2, 5], [2, -1], [6, 2], [-2, 2], [2, 6], [2, -2], [7, 2], [-3, 2],
+       [2, 7], [2, -3], [8, 2], [-4, 2], [2, 8], [2, -4], [9, 2], [-5, 2], [2, 9], [2, -5]].sort
+    end
+
+    it 'returns array of not filtered potential coordinates' do
+      expect(rook.moves.sort).to eql rook_moves
```
Comment (cs3b) [inline spec/lib/figure/rook_spec.rb]: you sure that sorting should be done on spec level?

if sort order doesn't matter, then take look at - https://www.relishapp.com/rspec/rspec-expectations/v/3-5/docs/built-in-matchers/contain-exactly-matcher

**Pull request merged**